api/routes/authRouter.js

```diff
@@ -0,0 +1,124 @@
+const express = require("express");
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcryptjs");
+
+const db = require("../../data/dbConfig");
+const { generateToken } = require("../../auth/authenticate");
+
+const router = express.Router();
+
+const secret =
+  process.env.JWT_SECRET ||
+  "add a .env file to root of project with the JWT_SECRET variable";
+
+// const generateToken = user => {
+//   const payload = {
+//     username: user.username
+//   };
+//   const options = {
+//     expiresIn: "1d",
+//     jwtid: 12345
+//   };
+//   return jwt.sign(payload, secret, options);
+// };
+
+router.post("/register", () => {
+  const user = req.body;
+
+  if (
+    !user.username ||
+    typeof user.username !== "string" ||
+    user.username === ""
+  ) {
+    res
+      .status(400)
+      .json({ error: "username must be included and must be a string" });
+  } else if (
+    !user.password ||
+    typeof user.password !== "string" ||
+    user.password === ""
+  ) {
+    res
+      .status(400)
+      .json({ error: "password must be included and must be a string" });
+  } else if (user.username.length > 255) {
+    res.status(400).json({ error: "username must not exceed 255 characters" });
+  } else if (user.password.length > 255) {
+    res.status(400).json({ error: "password must not exceed 255 characters" });
+  } else {
+    
+
+    // Hash password using bcrypt
+    const hash = bcrypt.hashSync(user.password, 15);
+    user.password = hash;
+
+    db("users")
+        .insert(user)
+        .then(ids => {
+            const id = ids[0];
+
+            db("users")
+                .where({ id })
```

Comment:
These `.where` and `.first` calls could be turned into another helper function to save you a couple lines. Maybe 
```
registered: id => db('users').where({id}).first()
```